stock_available_to_promise_release/models/stock_move.py

```diff
@@ -114,25 +114,38 @@ def _is_unrelease_allowed_on_origin_moves(self, origin_moves):
         )
 
     def _check_unrelease_allowed(self):
-        for move in self:
-            if not move.unrelease_allowed:
-                message = _(
-                    "You are not allowed to unrelease this move %(move_name)s.",
-                    move_name=move.display_name,
+        forbidden_moves = self.filtered(lambda m: not m.unrelease_allowed)
+        if not forbidden_moves:
+            return
+        message = _("You are not allowed to unrelease those deliveries:\n")
+
+        for picking, forbidden_moves_by_picking in groupby(
+            forbidden_moves, lambda m: m.picking_id
+        ):
```

Comment:
I don't quite like the naming of those variables, forbidden_moves_by_picking is a list of moves, but whatever.